dedupe/convenience.py

```diff
@@ -24,55 +24,94 @@ def consoleLabel(deduper): # pragma: no cover
     '''
 
     finished = False
+    has_previous = False
+    use_previous = False
     fields = unique(field.field
                     for field
                     in deduper.data_model.primary_fields)
 
 
-    while not finished :
-        n_match, n_distinct = (len(deduper.training_pairs['match']),
-                               len(deduper.training_pairs['distinct']))
-
-        uncertain_pairs = deduper.uncertainPairs()
-
+    buffer_len = 1 # Max number of previous operations
+    examples_buffer = []
+    record_pair_buffer = [] 
```

Comment:
Might be cleaner to make `example_buffer` a list of tuples of labels and the record_pair. You can turn these into the format that markPairs expects at the last moment. 

If you did this you wouldn't need two buffers.